relayer/src/link/cli.rs

```diff
@@ -51,23 +58,48 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> Link<ChainA, ChainB> {
         )
         .entered();
 
-        self.a_to_b.build_recv_packet_and_timeout_msgs(None)?;
+        // Relaying on a non-zero connection delay requires (indefinite) blocking
+        // to wait for the connection delay to pass.
+        // We do not support this in interactive mode.
+        if !self.a_to_b.channel().connection_delay.is_zero() {
+            error!(
+                "relaying on a non-zero connection delay path is not supported in interactive mode"
+            );
+            panic!("please use the passive relaying mode (`hermes start`)");
+        }
 
-        let mut results = vec![];
+        // Find the sequence numbers of unreceived packets
+        let (sequences, src_response_height) = unreceived_packets(
+            self.a_to_b.dst_chain(),
+            self.a_to_b.src_chain(),
+            &self.a_to_b.path_id,
+        )
+        .map_err(LinkError::supervisor)?;
 
-        // Block waiting for all of the scheduled data (until `None` is returned)
-        while let Some(odata) = self.a_to_b.fetch_scheduled_operational_data()? {
-            let mut last_res = self
-                .a_to_b
-                .relay_from_operational_data::<relay_sender::SyncSender>(odata)?;
-            results.append(&mut last_res.events);
+        if sequences.is_empty() {
+            return Ok(vec![]);
+        }
+
+        info!("unreceived packets found: {} ", sequences.len());
+
+        // Relay
+        let mut results = vec![];
+        for events_chunk in query_packet_events_with(
+            &sequences,
+            src_response_height,
+            self.a_to_b.src_chain(),
+            &self.a_to_b.path_id,
+            query_send_packet_events,
+        ) {
+            let mut last_events = self.a_to_b.relay_from_events(events_chunk)?;
```

Comment:
Mentioned during a weekly ibc-rs call, but is it within the scope of this PR to attempt to async-ify the relaying of events here?

Reply:
>Not sure. This PR is becoming way too complex. I would concentrate on the minimal fixes for the original issue without doing major rework.

+1

I would like to pursue async-ification when it's off the critical path of fixing bugs or adding important features (as in this PR).